### PR TITLE
Fix indigo build issues

### PR DESCRIPTION
**Pull request by jrgnicho**
@shaun-edwards, the changes in this PR fix the build errors in ros-indigo.  You could either merge this and/or use it as a departure point for the **indigo-devel** branch.

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
fyi: `cmake_modules` is fully (bw-)compatible with Hydro.

File: descartes_core/package.xml [hunk 1]
```diff
@@ -17,11 +17,13 @@
   <build_depend>console_bridge</build_depend>
   <build_depend>moveit_core</build_depend>
   <build_depend>roscpp</build_depend>
+  <build_depend>cmake_modules</build_depend>
 
   <run_depend>boost</run_depend>
   <run_depend>console_bridge</run_depend>
   <run_depend>moveit_core</run_depend>
   <run_depend>roscpp</run_depend>
+  <run_depend>cmake_modules</run_depend>
```
Comment (gavanderhoorn) [inline descartes_core/package.xml]: I don't believe a `run_depend` is needed with `cmake_modules`. It's a build-only thing.

Comment (jrgnicho) [inline descartes_core/package.xml]: Ok, I'll list cmake_modules as a build_depend only

**Comment by jrgnicho** [comment]
The latest commits mainly removed the **console_bridge** dependencies.  This branch builds in both hydro and indigo.  @gavanderhoorn I tried removing the **cmake_modules** run dependency but it game me the following build error:

```
CMake Error at /opt/ros/indigo/share/catkin/cmake/catkin_package.cmake:217 (message):
  catkin_package() DEPENDS on the catkin package 'cmake_modules' which must
  therefore be listed as a run dependency in the package.xml
```

**Comment by gavanderhoorn** [comment]
@jrgnicho: yes, but `cmake_modules` should not be added to `CATKIN_DEPENDS` either, afaik. See [catkin_package() - CATKIN_DEPENDS](http://docs.ros.org/hydro/api/catkin/html/dev_guide/generated_cmake_api.html#catkin-package) on the _Extracted CMake API reference_ page:

> [..] Therefore their INCLUDE_DIRS and LIBRARIES will be appended to ours. [..]

`cmake_modules` doesn't have any headers or libraries.

And you can also just `find_package(..)` it on its own, no need to add it to the `find_package(catkin ..)` line.

**Comment by jrgnicho** [comment]
@gavanderhoorn The last commits removes the cmake_modules run dependencies.  It still builds on both hydro and indigo

File: descartes_core/CMakeLists.txt [hunk 2]
```diff
@@ -3,13 +3,12 @@ project(descartes_core)
...
 add_definitions(-std=gnu++0x)
 find_package(catkin REQUIRED COMPONENTS
-  console_bridge
   moveit_core
   roscpp
+  cmake_modules
 )
 
 find_package(Boost REQUIRED)
-find_package(Eigen REQUIRED)
```
Comment (gavanderhoorn) [inline descartes_core/CMakeLists.txt:12]: Why is this removed if `Eigen` is listed as a `DEPENDS` in `catkin_package(..)` further on (and `Eigen_INCLUDE_DIRS` is also referenced)?

Also: if you don't use `Eigen` in this package, then it would seem there is also no need for `cmake_modules`?

File: descartes_planner/CMakeLists.txt [hunk 3]
```diff
@@ -24,9 +22,10 @@ catkin_package(
   CATKIN_DEPENDS
     roscpp
     moveit_core
+    descartes_core
+    descartes_trajectory
   DEPENDS
     Boost
-    console_bridge
     Eigen
```
Comment (gavanderhoorn) [inline descartes_planner/CMakeLists.txt]: See earlier comment.

File: descartes_trajectory/CMakeLists.txt [hunk 4]
```diff
@@ -2,14 +2,12 @@ cmake_minimum_required(VERSION 2.8.3)
...
 add_definitions(-std=gnu++0x)
 find_package(catkin REQUIRED COMPONENTS
-  console_bridge
   descartes_core
   moveit_core
   roscpp
 )
 
 find_package(Boost REQUIRED)
-find_package(Eigen REQUIRED)
```
Comment (gavanderhoorn) [inline descartes_trajectory/CMakeLists.txt:12]: See earlier comment.

**Comment by gavanderhoorn** [comment]
If I understand things correctly, the `boost` and `eigen` keys should be added as `build_depend`s, and a `run_depend` on `boost` is necessary (but only if any of its libraries are used).

For `Eigen` there are actually three `rosdep` keys: [eigen](http://rosindex.github.io/d/eigen/), [eigen2](http://rosindex.github.io/d/eigen2/) and [eigen3](http://rosindex.github.io/d/eigen3/). For `Boost` there is only one: [boost](http://rosindex.github.io/d/boost/).

File: descartes_core/CMakeLists.txt (hunk 2 above)
Comment (jrgnicho) [inline descartes_core/CMakeLists.txt:12]: The Eigen library is used in this package.  Wouldn't **find(catkin REQUIRED COMPONENTS cmake_modudes ...)** include Eigen as well;  in which case calling find_package(Eigen ...) would be redundant.  If this is so then adding Eigen_INCLUDE_DIRS or Eigen under DEPENDS isn't required either.

Comment (gavanderhoorn) [inline descartes_core/CMakeLists.txt:12]: Nope. You have to `find_package(Eigen ..)` yourself. `cmake_modules` just provides the find script.

Comment (jrgnicho) [inline descartes_core/CMakeLists.txt:12]: I'm surprised that omitting it didn't lead to any build errors.  I'll place them back in the CMakeList files.

Comment (gavanderhoorn) [inline descartes_core/CMakeLists.txt:12]: Could be that you have Eigen installed in the default location (ie: `/usr/include`), which is always on the include path, or some other dependency already did a `find_package(Eigen ..)` somewhere.

**Comment by jrgnicho** [comment]
Last commit reinstated find_package(Eigen ...) and squashed all previous commits into one.

**Comment by gavanderhoorn** [comment]
Looking good. We don't want to add a `build_depend` on `Eigen` to the manifest? See earlier comment for the `rosdep` keys.

**Comment by jrgnicho** [comment]
Would omitting it from the package.xml files break anything?

**Comment by gavanderhoorn** [comment]
It would on systems with no `Eigen` installed. A `rosdep check ..` would not report `Eigen` not being installed then.

I don't really see a reason to not explicitly state all dependencies though.

**Comment by jrgnicho** [comment]
build depends for Eigen have been added to the package.xml files

**Comment by gavanderhoorn** [comment]
I think this is good to go.

+1 from me.
